test/modules/XMLHttpRequest/compliance/xhr-event-handlers.browser.test.ts

```diff
@@ -0,0 +1,55 @@
+import { HttpServer } from '@open-draft/test-server/http'
+import { test, expect } from '../../../playwright.extend'
+import { useCors } from '../../../helpers'
+
+const httpServer = new HttpServer((app) => {
+  app.use(useCors)
+  app.get('/resource', (req, res) => {
+    res.status(200).send('hello')
+  })
+})
+
+test.beforeAll(async () => {
+  await httpServer.listen()
+})
+
+test.afterAll(async () => {
+  await httpServer.close()
+})
+
+test('onloadend handler is called', async ({ page, loadExample }) => {
+  await loadExample(require.resolve('./xhr-event-handlers.browser.runtime.js'))
+
+  const { request, calls } = await page.evaluate(async (url) => {
+    const calls = {
+      loadEndHandler: 0,
+      loadEndListener: 0,
+    }
+
+    const xhr = new XMLHttpRequest()
+    xhr.open('GET', url)
+    xhr.onloadend = () => calls.loadEndHandler++
+    xhr.addEventListener('loadend', () => calls.loadEndListener++)
+
+    await Promise.all([
+      new Promise((resolve) => {
+        const resolveDelayed = () => setTimeout(resolve, 1000)
+        xhr.addEventListener('error', resolveDelayed)
+        xhr.addEventListener('load', resolveDelayed)
+      }),
+      xhr.send(null),
```

Comment:
Since `xhr.send` doesn't return a Promise, is there a reason to put these two in a single `Promise.all`?

Reply:
I wanted to avoid adding the event listeners after the `send` call.

```ts
xhr.send(null)

await new Promise((resolve) => {
  const resolveDelayed = () => setTimeout(resolve, 1000)
  xhr.addEventListener('error', resolveDelayed)
  xhr.addEventListener('load', resolveDelayed)
})
```

seems to work fine though. Do think that’s fine or would you rather see a more specific version like:

```ts
xhr.send(null)

const finishedPromise = new Promise((resolve) => {
  const resolveDelayed = () => setTimeout(resolve, 1000)
  xhr.addEventListener('error', resolveDelayed)
  xhr.addEventListener('load', resolveDelayed)
})

xhr.send(null)
await finishedPromise
```